src/builder.rs

```diff
@@ -127,16 +127,17 @@ impl TransactionBuilder {
         &mut self,
         secret: SecretKey,
         sender: fuel_types::Address,
-        recipient: fuel_types::Address,
         nonce: Word,
         amount: Word,
         data: Vec<u8>,
     ) -> &mut Self {
         let pk = secret.public_key();
-
         self.sign_keys.push(secret);
+
+        let recipient = Input::owner(&pk);
```

Comment:
This is a bit confusing. Should the recipient be the owner?

Reply:
Basically yes, https://fuellabs.slack.com/archives/C01L1HUJSL8/p1661889305009839?thread_ts=1661889033.348139&cid=C01L1HUJSL8